src/dbnode/storage/entry.go

```diff
@@ -292,16 +379,37 @@ func (entry *Entry) TryMarkIndexGarbageCollected(reconciled, unreconciled tally.
 	// marks this GCed bool.
 	e.IndexGarbageCollected.Store(true)
 
-	// Was reconciled if the entry retrieved from the shard differs from the current.
 	if e != entry {
-		reconciled.Inc(1)
+		entry.metrics.gcNeedsReconcile.Inc(1)
 	} else {
-		unreconciled.Inc(1)
+		entry.metrics.gcNoReconcile.Inc(1)
 	}
 
 	return true
 }
 
+// TryReconcileDuplicates attempts to reconcile the index states of this entry.
+func (entry *Entry) TryReconcileDuplicates() {
+	// Since series insertions + index insertions are done separately async, it is possible for
+	// a series to be in the index but not have data written yet, and so any series not in the
+	// lookup yet we cannot yet consider empty.
+	e, _, err := entry.Shard.TryRetrieveSeriesAndIncrementReaderWriterCount(entry.ID)
```

Comment:
nit: can probably share this logic w/ TryMarkGC w/ a private func - just because this pattern of retrieving series, dec, and merge, are all in both places.

Reply:
It's a bit annoying since the dec is a `defer` so we can't run that from a private function, and because of that it functionally ends up having a fairly similar amount of boilerplate either way; may leave it as is